lib/src/sysutils.cpp

```diff
@@ -0,0 +1,42 @@
+#include <cstdio>
+#include <string>
+#include <cassert>
+#include <stdexcept>
+
+#include <dlisio/sysutils.hpp>
+
+#ifdef _WIN32
+    #include <windows.h>
+#endif
+
+namespace dlisio {
+
+std::FILE* sysopen(const std::string& path) noexcept (false) {
```

Comment:
Would prefer just `open`, and same for the file names. Alternatively, put it in stream.hpp to avoid file clutter